x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/datafeed_step/components/frequency/frequency_input.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useContext, useEffect } from 'react';
+import { EuiFieldText } from '@elastic/eui';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { Description } from './description';
+import { calculateDatafeedFrequencyDefaultSeconds } from '../../../../../../../../common/util/job_utils';
+
+export const FrequencyInput: FC = () => {
+  const { jobCreator, jobCreatorUpdate, jobValidator, jobValidatorUpdated } = useContext(
+    JobCreatorContext
+  );
+  const [validation, setValidation] = useState(jobValidator.frequency);
+  const [frequency, setFrequency] = useState(
+    jobCreator.frequency === null ? '' : jobCreator.frequency
```

Comment:
Might be worth creating a function for `jobCreator.frequency === null ? '' : jobCreator.frequency` it's used similarly multiple times in this file.